### PR TITLE
Add initial area charts

### PR DESCRIPTION
**Pull request by Kongkille**
## Issue
https://linear.app/electricitymaps/issue/ELE-1342/[ui]-panel-zone-details-charts

## Description

Adds
- CarbonChart
- BreakdownChart
- PriceChart

All of these use areagraph as the core component. I've added it with as minimal changes as possible for now

### Preview

https://user-images.githubusercontent.com/45588799/203585108-8a52c5b5-2083-448c-9431-68d65c348c8b.mov

### CODE REVIEW

File: web/src/types.ts [hunk 1]
```diff
@@ -56,9 +56,35 @@ export interface ZoneOverview {
...
+  totalco2intensity: number;
+  totalCo2Import: number;
+  totalCo2Discharge: number;
+  totalCo2Production: number;
+  totalProduction: number;
+  totalImport: number;
+  totalDischarge: number;
+  storage: { [key in StorageType]: number };
+  price?: {
+    value: number;
```
Comment (madsnedergaard) [inline web/src/types.ts:86]:
```suggestion
    value: number;
    currency: string;
```

**Review by madsnedergaard — APPROVED** [review]
This is amazing, impressive that you got it working so fast despite it being such messy code! 🚀 
Some minor comments, but nothing big so I'll approve :)

**Comment by Kongkille** [comment]
Thanks for the review, I'll incorporate your comments into the next iteration :)

File: web/src/features/time/TimeAxis.tsx [hunk 2]
```diff
@@ -82,26 +95,28 @@ function TimeAxis({ selectedTimeAggregate, datetimes, isLoading }: TimeAxisProps
...
   }
 
-  const displayLive = selectedTimeAggregate === TimeAverages.HOURLY;
-  // Required to render the ticks in the proper width on resize
-  const scale = getTimeScale(width, datetimes[0], datetimes[datetimes.length - 1]);
+  scale = getTimeScale(
+    scaleWidth ?? width,
+    datetimes[0],
+    datetimes[datetimes.length - 1]
+  );
```
[the remark this answers is not in the document]
Comment (Kongkille) [reply web/src/features/time/TimeAxis.tsx:102]: Removing from props,

And yeah the timeaxis component has already become a bit of a mess again. When i rewrote it last week, I had still thought we went for recharts so I hadn't thought it should be reused by the svg components -.-

Maybe we can fix the whole resize stuff later

**Pull request merged**